python/lsst/ip/isr/isrTask.py

```diff
@@ -1770,6 +1771,115 @@ def run(self, ccdExposure, *, camera=None, bias=None, linearizer=None,
+            else:
+                # Try then with the amplifier gain.
+                # We already have a detector at this point. If there was no
+                # detector to beging with, one would have been created with
+                # self.config.gain and self.config.noise. Same comment
+                # applies for the noise block below.
+                gain = amp.getGain()
+
+            # Check if the gain up to this point differs from the
+            # gain in bfGains. If so, raise or warn, accordingly.
+            if not boolGainMismatch and bfGains is not None:
+                bfGain = bfGains[ampName]
+                if not math.isclose(gain, bfGain, rel_tol=1e-4):
+                    if self.config.doRaiseOnCalibMismatch:
+                        raise RuntimeError("Gain mismatch for det %s amp %s: "
+                                           "(gain (%s): %s, bfGain: %s)",
+                                           detName, ampName, gainProvenanceString,
+                                           gain, bfGain)
+                    else:
+                        self.log.warning("Gain mismatch for det %s amp %s: "
+                                         "(gain (%s): %s, bfGain: %s)",
+                                         detName, ampName, gainProvenanceString,
+                                         gain, bfGain)
+                    boolGainMismatch = True
+
+            # Noise:
+            # Try first with the empirical noise from the overscan.
+            noiseProvenanceString = "amp"
+            if self.config.doEmpiricalReadNoise and overscanResults is not None:
+                noiseProvenanceString = "serial overscan"
+                if isinstance(overscanResults.residualSigma, float):
+                    # Only serial overscan was run
+                    noise = overscanResults.residualSigma
+                else:
+                    # Both serial and parallel overscan were
+                    # run.  Only report noise from serial here.
+                    noise = overscanResults.residualSigma[0]
+            elif self.config.usePtcReadNoise:
+                # Try then with the PTC noise.
+                if ptcDataset is None:
+                    raise RuntimeError("No ptcDataset provided to use PTC noise.")
+                else:
+                    noise = ptcDataset.noise[amp.getName()]
+                    noiseProvenanceString = "ptc"
+                    self.log.info("Using noise from Photon Transfer Curve.")
+            else:
+                # Finally, try with the amplifier noise.
+                # We already have a detector at this point. If there
+                # was no detector to beging with, one would have
+                # been created with self.config.gain and
+                # self.config.noise.
+                noise = amp.getReadNoise()
+
+            self.log.info("%s - Noise provenance: %s, Gain provenance: %s",
```

Comment:
I think this (and the metadata entries) can be pulled out of the loop.  All amps should follow the same logic, so all amps should have the same provenance, right?